sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.netstandard2.0.cs

```diff
@@ -120,3 +120,73 @@ public ProvisionableTemplateAttribute(string resourceName) { }
         public string ResourceName { get { throw null; } }
     }
 }
+namespace System.ClientModel
+{
+    public partial interface IAuthorizationCodeFlowContext : System.ClientModel.IScopedFlowContext, System.ClientModel.ITokenContext
+    {
+        System.Uri AuthorizationUri { get; }
+        System.Uri RefreshUri { get; }
+        System.Uri TokenUri { get; }
+    }
+    public partial interface IClientCredentialsFlowContext : System.ClientModel.IScopedFlowContext, System.ClientModel.ITokenContext
+    {
+        System.Uri RefreshUri { get; }
+        System.Uri TokenUri { get; }
+    }
+    public partial interface IImplicitFlowContext : System.ClientModel.IScopedFlowContext, System.ClientModel.ITokenContext
+    {
+        System.Uri AuthorizationUri { get; }
+        System.Uri RefreshUri { get; }
+    }
+    public partial interface IPasswordFlowContext : System.ClientModel.IScopedFlowContext, System.ClientModel.ITokenContext
+    {
+        System.Uri RefreshUri { get; }
+        System.Uri TokenUri { get; }
+    }
+    public partial interface IScopedFlowContext : System.ClientModel.ITokenContext
+    {
+        string[] Scopes { get; }
+        object CloneWithAdditionalScopes(string[] additionalScopes);
+    }
+    public partial interface ITokenContext
+    {
+    }
+    public partial interface ITokenProvider
+    {
+        object CreateContext(System.Collections.Generic.IReadOnlyDictionary<string, object> properties);
+        System.ClientModel.Token GetAccessToken(object context, System.Threading.CancellationToken cancellationToken);
+        System.Threading.Tasks.ValueTask<System.ClientModel.Token> GetAccessTokenAsync(object context, System.Threading.CancellationToken cancellationToken);
+    }
+    public abstract partial class RefreshableToken : System.ClientModel.Token
+    {
+        public RefreshableToken(string tokenValue, string tokenType, System.DateTimeOffset expiresOn, System.DateTimeOffset? refreshOn = default(System.DateTimeOffset?)) : base (default(string), default(string), default(System.DateTimeOffset), default(System.DateTimeOffset?)) { }
+        public abstract System.Threading.Tasks.Task RefreshAsync(System.Threading.CancellationToken cancellationToken);
+    }
+    public partial class Token
```

Comment:
We should chat about whether `Token` and `RefreshableToken` are classes or interfaces.  Classes are strongly preferred, but interfaces might make it easier to tie into other auth libraries (i.e., they could still explicitly implement `IToken` if their primary credential type already has a base class).